services/brig/test/integration/API/Team.hs

```diff
@@ -152,6 +154,40 @@ testInvitationEmail brig = do
   let invite = stdInvitationRequest invitee (Name "Bob") Nothing Nothing
   void $ postInvitation brig tid inviter invite
 
+testInvitationEmailLookup :: Brig -> Http ()
+testInvitationEmailLookup brig = do
+  email <- randomEmail
+  -- expect no invitation to be found for an email before that person is invited
+  headInvitationByEmail brig email 404
+  (uid, tid) <- createUserWithTeam brig
+  let invite = stdInvitationRequest email (Name "Bob") Nothing Nothing
+  void $ postInvitation brig tid uid invite
+  -- expect an invitation to be found querying with email after invite
+  headInvitationByEmail brig email 200
+  (uid2, tid2) <- createUserWithTeam brig
+  let invite2 = stdInvitationRequest email (Name "Bob2") Nothing Nothing
+  void $ postInvitation brig tid2 uid2 invite2
+  -- expect a 409 conflict result for a second team inviting the same user
+  headInvitationByEmail brig email 409
+
+testInvitationEmailLookupRegister :: Brig -> Http ()
+testInvitationEmailLookupRegister brig = do
+  email <- randomEmail
+  (owner, tid) <- createUserWithTeam brig
+  let invite = stdInvitationRequest email (Name "Bob") Nothing Nothing
+  void $ postInvitation brig tid owner invite
+  inv :: Invitation <- responseJsonError =<< postInvitation brig tid owner invite
+  -- expect an invitation to be found querying with email after invite
+  headInvitationByEmail brig email 200
+  void $ registerInvite brig tid inv email
+  -- expect a 404 after invitation has been used.
+  headInvitationByEmail brig email 404
```

Comment:
Is this removed synchronously?  Otherwise you may want to wrap this in a call to [retrying or similar](https://hoogle.zinfra.io/file/root/.stack/snapshots/x86_64-linux/01af7b6eb1c680941d3ef08786685b31d2470c8b36e981c7bdad15d91f453def/8.6.5/doc/retry-0.8.1.0/Control-Retry.html) here.

Reply:
Yes, it's removed synchronously in both the `Brig/Team/API.hs` deleteInvitation(s) endpoints, as well as on user creation after user activation - all synchronously. So no need for retrying wrapping.